### PR TITLE
Add nft support

### PR DESCRIPTION
**Pull request by mukulljangid**
## High Level Overview of Change
Adds NFT Support.

Also makes the `definitions.json` file portable from JS to Python, which will be useful moving forward. This happens by remapping the python class `SerializedDict` to `STObject` and `SerializedList` to `STArray`.

Also makes camel-to-snake case conversion generic. There should no longer be any exceptions.

Also uses a list of easier-to-read and easier-to-amend exceptions for snake-to-camel case conversion. This should make it easier to add new exceptions.

Also improves type hints around these changes.

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### CODE REVIEW

File: xrpl/models/transactions/nftoken_cancel_offer.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    The transaction can be executed by the account that originally created
+    the NFTokenOffer, the account in the `Recipient` field of the NFTokenOffer
+    (if present), or any account if the NFTokenOffer has an `Expiration` and
+    the NFTokenOffer has already expired.
+    """
+
+    token_ids: List[str] = REQUIRED  # type: ignore
+    """
+    TODO - unclear if this is token_ids or token_offers
```
Comment (ledhed2222) [inline xrpl/models/transactions/nftoken_cancel_offer.py:29]: TODO

**Review by ledhed2222 — APPROVED** [review]

**Comment by ledhed2222** [comment]
TODO - need to verify once more all the fields in binary codec definitions, especially the values of the new error codes

File: xrpl/models/amounts/amount.py [hunk 2]
```diff
@@ -34,3 +34,18 @@ def is_issued_currency(amount: Amount) -> bool:
         Whether the amount is an issued currency value.
     """
     return isinstance(amount, IssuedCurrencyAmount)
+
+
+def value(amount: Amount) -> float:
```
Comment (mvadari) [inline xrpl/models/amounts/amount.py:39]: can this have a more descriptive name?

File: xrpl/core/binarycodec/definitions/field_instance.py [hunk 3]
```diff
@@ -46,8 +46,8 @@ def _get_type_by_name(name: str) -> Type[SerializedType]:
         "Hash160": Hash160,
         "Hash256": Hash256,
         "PathSet": PathSet,
-        "SerializedList": SerializedList,
-        "SerializedDict": SerializedDict,
+        "STArray": SerializedList,
+        "STObject": SerializedDict,
```
Comment (khancode) [inline xrpl/core/binarycodec/definitions/field_instance.py:50]: What does ST stand for?

File: xrpl/models/transactions/account_set.py [hunk 4]
```diff
@@ -172,3 +184,37 @@ def _transfer_rate_error(self: AccountSet) -> Optional[str]:
...
+        return None
+
+    def _get_minter_error(self: AccountSet) -> Optional[str]:
+        if (
+            self.set_flag != AccountSetFlag.ASF_AUTHORIZED_MINTER
+            and self.minter is not None
+        ):
+            return """Will not set the minter unless \
+                AccountSetFlag.ASF_AUTHORIZED_MINTER is set\
+                """
```
Comment (mvadari) [inline xrpl/models/transactions/account_set.py:207]: The rest of the repo uses the syntax
```python
(
  "stuff"
  "more stuff"
)
```
for longer strings

File: xrpl/models/requests/request.py [hunk 5]
```diff
@@ -147,17 +146,20 @@ def get_method(cls: Type[Request], method: str) -> Type[Request]:
...
-        return request_models.GenericRequest
+        # special case for NoRippleCheck and NFT methods
+        if method == RequestMethod.NO_RIPPLE_CHECK:
+            return xrpl.models.requests.NoRippleCheck
+        if method == RequestMethod.ACCOUNT_NFTS:
+            return xrpl.models.requests.AccountNFTs
+        if method == RequestMethod.NFT_BUY_OFFERS:
+            return xrpl.models.requests.NFTBuyOffers
+        if method == RequestMethod.NFT_SELL_OFFERS:
+            return xrpl.models.requests.NFTSellOffers
```
Comment (khancode) [inline xrpl/models/requests/request.py:157]: Not sure if we can do this in Python but can we reduce this to `return xrpl.models.requests[RequestMethod.METHOD_NAME]`

Comment (khancode) [reply xrpl/models/requests/request.py:157]: Oh wait, there's more logic after this conditional check. Nevermind :)

**Review by mvadari — APPROVED** [review]

**Review by khancode — APPROVED** [review]
LGTM 👌

**Pull request merged**